controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -347,32 +349,38 @@ private boolean validateCapabilities() {
             for (Map.Entry<Step, Set<CapabilityId>> entry : missingForStep.entrySet()) {
                 Step step = entry.getKey();
                 ModelNode response = step.response;
-                // only overwrite reponse failure-description if there isn't one
-                StringBuilder msg = response.hasDefined(FAILURE_DESCRIPTION)
-                        ? null
-                        : new StringBuilder(ControllerLogger.ROOT_LOGGER.requiredCapabilityMissing());
-                StringBuilder bootMsg = isBooting() || ignoreFailures
-                        ? new StringBuilder(ControllerLogger.ROOT_LOGGER.requiredCapabilityMissing(step.address.toCLIStyleString()))
-                        : null;
+                StringBuilder unsupportedExpressionsMsg = new StringBuilder();
+                StringBuilder capabilityMissingMsg = new StringBuilder();
                 for (CapabilityId id : entry.getValue()) {
-                    String formattedCapability = ignoreContext
-                            ? ControllerLogger.ROOT_LOGGER.formattedCapabilityName(id.getName())
-                            : ControllerLogger.ROOT_LOGGER.formattedCapabilityId(id.getName(), id.getScope().getName());
-                    Set<PathAddress> possiblePoints = managementModel.getCapabilityRegistry().getPossibleProviderPoints(id);
-                    if (msg != null) {
-                        msg = appendPossibleProviderPoints(msg, formattedCapability, possiblePoints);
-                    }
-                    if (bootMsg != null) {
-                        bootMsg = appendPossibleProviderPoints(bootMsg, formattedCapability, possiblePoints);
+                    if (id.getName().matches(UNSUPPORTED_EXPRESSION_PATTERN)) {
```

Comment:
What kind of performance overhead will this cause?  This pattern is going to have to be recompiled for every ID encountered.